src/inmanta/deploy/scheduler.py

```diff
@@ -422,7 +444,11 @@ async def report_resource_state(
                         task = Deploy(resource=resource)
                         assert task in self._work.agent_queues.in_progress
                         priority = self._work.agent_queues.in_progress[task]
-                        self._work.deploy_with_context(event_listeners, priority=priority, deploying=set())
+                        task_with_reason = [e for e in self._work.agent_queues.in_progress.keys() if e == task][0]
+                        assert task_with_reason.reason is not None
```

Comment:
I would like to see some explanation if we reach this assert, it will make it a bit easier for debugging. Same for the others

Reply:
Either a comment or the message of the assertion error